Tasks/DotNetCoreInstallerV1/versionInstaller.ts

```diff
@@ -0,0 +1,132 @@
+        }
+
+        //todo: in case of failure due to: unable to acquire lock or timeout, 3 retries to install.
+    }
+
+    public isVersionInstalled(version: string): boolean {
+        var isInstalled: boolean = false;
+        if (this.packageType == "sdk") {
+            isInstalled = toolLib.isExplicitVersion(version) && fs.existsSync(path.join(this.installationPath, "sdk", version)) && fs.existsSync(path.join(this.installationPath, "sdk", `${version}.complete`));
+        }
+        else {
+            isInstalled = toolLib.isExplicitVersion(version) && fs.existsSync(path.join(this.installationPath, "shared", "Microsoft.NETCore.App", version)) && fs.existsSync(path.join(this.installationPath, "shared", "Microsoft.NETCore.App", `${version}.complete`));
+        }
+
+        isInstalled ? console.log(tl.loc("VersionFoundInToolCache")) : console.log(tl.loc("VersionNotFoundInToolCache", version));
+        return isInstalled;
+    }
+
+    private createInstallationCompleteFile(versionInfo: VersionInfo): void {
+        let version = versionInfo.version;
+        tl.debug(tl.loc("CreatingInstallationCompeleteFile", version, this.packageType));
+        // always add for runtime as it is installed with SDK as well.
+        var pathToVersionCompleteFile: string = "";
+        if (this.packageType == "sdk") {
+            pathToVersionCompleteFile = path.join(this.installationPath, "sdk");
+            fs.writeFileSync(path.join(pathToVersionCompleteFile, `${version}.complete`), `{ "version": "${version}" }`);
+        }
+
+        pathToVersionCompleteFile = path.join(this.installationPath, "shared", "Microsoft.NETCore.App");
+        fs.writeFileSync(path.join(pathToVersionCompleteFile, `${VersionInfo.getRuntimeVersion(versionInfo)}.complete`), `{ "version": "${VersionInfo.getRuntimeVersion(versionInfo)}" }`);
+    }
+
+    private isLatestInstalledVersion(version: string): boolean {
+        var pathTobeChecked = this.packageType == "sdk" ? path.join(this.installationPath, "sdk") : path.join(this.installationPath, "host", "fxr");
+        if (!fs.existsSync(pathTobeChecked)) {
+            throw tl.loc("PathNotFoundException", pathTobeChecked);
+        }
+
+        var allEnteries: string[] = fs.readdirSync(pathTobeChecked).map(name => path.join(pathTobeChecked, name));
+        var folderPaths: string[] = allEnteries.filter(element => fs.lstatSync(element).isDirectory());
+        var isLatest: boolean = folderPaths.findIndex(folderPath => utils.versionCompareFunction(path.basename(folderPath), version) > 0) < 0;
+        var filePaths: string[] = allEnteries.filter(element => !fs.lstatSync(element).isDirectory());
+        isLatest = isLatest && filePaths.findIndex(filePath => utils.versionCompareFunction(this.getVersionCompleteFileName(path.basename(filePath)), version) > 0) < 0;
+
+        isLatest ? tl.debug(tl.loc("VersionIsLocalLatest")) : tl.debug(tl.loc("VersionIsNotLocalLatest"));
+        return isLatest;
```

Comment:
Will handle presence of files and folders not in version format, and ignore those